go/adbc/driver/snowflake/statement.go

```diff
@@ -0,0 +1,572 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package snowflake
+
+import (
+	"context"
+	"database/sql/driver"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/snowflakedb/gosnowflake"
+)
+
+const (
+	OptionStatementQueueSize = "adbc.flight.sql.rpc.queue_size"
```

Comment:
Do we want to reuse the exact same option name? (Maybe that's for the better.)

Reply:
hmm. should we change the option name to be more generic?

Reply:
Possibly something like `adbc.???.fetch_queue_size` (if there's a namespace we're already using that can go in to `???`)

Reply:
updated to `adbc.sql.results.rpc.queue_size`

Reply:
Kind of bikeshedding, but maybe drop the 'sql'? 'adbc.rpc.result_queue_size'?

Reply:
sounds good, i'll update it